packages/react-ui/src/widgets/TimeSeriesWidgetUI/TimeSeriesWidgetUI.js

```diff
@@ -15,18 +17,26 @@ import { PropTypes } from 'prop-types';
 import { GroupDateTypes, getMonday } from '@carto/react-core';
 
 const FORMAT_DATE_BY_STEP_SIZE = {
-  [GroupDateTypes.DAYS]: daysCurrentDateRange,
-  [GroupDateTypes.WEEKS]: weeksCurrentDateRange
+  [GroupDateTypes.YEARS]: yearCurrentDateRange,
+  [GroupDateTypes.MONTHS]: monthsCurrentDateRange,
+  [GroupDateTypes.WEEKS]: weeksCurrentDateRange,
+  [GroupDateTypes.DAYS]: daysCurrentDateRange
 };
 
-const FORMAT_TIMEFRAME_BY_STEP_SIZE = {
-  [GroupDateTypes.DAYS]: daysCurrentDateRange,
-  [GroupDateTypes.WEEKS]: daysCurrentDateRange
+const FORMAT_TIME_WINDOW_BY_STEP_SIZE = {
+  [GroupDateTypes.YEARS]: yearCurrentDateRange,
+  [GroupDateTypes.MONTHS]: monthsCurrentDateRange,
+  [GroupDateTypes.WEEKS]: daysCurrentDateRange,
+  [GroupDateTypes.DAYS]: daysCurrentDateRange
 };
 
-const TIMEFRAME_STEP_BY_STEP_SIZE = {
-  [GroupDateTypes.DAYS]: 60 * 60 * 24,
-  [GroupDateTypes.WEEKS]: 60 * 60 * 24
+const TIME_WINDOW_STEP_BY_STEP_SIZE = {
+  [GroupDateTypes.YEARS]: 60 * 60 * 24 * 7,
```

Comment:
YEARS look related to 1 WEEK, @Clebal are these values correct?

Reply:
In other piece of the code you use: [GroupDateTypes.YEARS]: 60 * 60 * 24 * 7 * 30 * 12 so...